src/index.ts

```diff
@@ -1,36 +1,101 @@
-import "reflect-metadata"
-import { AppDataSource } from "./data-source"
-import { User } from "./entity/User"
+import 'reflect-metadata';
 import { ApolloServer } from '@apollo/server';
 import { startStandaloneServer } from '@apollo/server/standalone';
+import { AppDataSource } from './data-source';
+import { User } from './entity/User';
+import 'bcrypt';
 
-AppDataSource.initialize().then(async () => {
-
-    const user = new User()
-    user.firstName = "3"
-    user.lastName = ""
-    user.age = 20
-
-    await AppDataSource.manager.save(user)
-})
-
+//GraphQl Schema
+//para colocar input (input NomeInput)
 const typeDefs = `#graphql
-    type Query {
+
+  type Query{
     hello: String
+  }
+
+  #input obrigatório !
+  type Mutation{
+    CreateUser(input: CreateUserInput!): User
 }
+  
+  input CreateUserInput{
+    name: String!
+    email: String!
+    password: String!
+    birthDate: String!
+  }
+
+   type User {
+    id: ID
+    name: String
+    email: String
+    password: String
+    birthDate: String
+  }
 `;
+//interface para conectar args com os inputs
+//typescript como default é obrigatório, para ser "opcional" -> ?
+interface CreateUserInput {
+  input: {
+    name: string;
+    email: string;
+    password: string;
+    birthDate: string;
+  };
+}
+
+const resolvers = {
+  Query: {
+    hello: () => 'Hello World',
+  },
+  Mutation: {
+    CreateUser: async (_: any, args: CreateUserInput) => {
+      //busca do email - busca no banco se já existe um email igual
+      const storageUsers = await AppDataSource.manager.findOne(User, { where: { email: args.input.email } });
+      if (storageUsers) {
+        throw new Error('Email já registrado');
+      }
+      //validação da senha - 6 caracteres, com 1 letra e 1 digito
+      const validPassword = args.input.password;
+      if (validPassword.length < 6 || !/\d/.test(validPassword) || !/[a-zA-Z]/.test(validPassword)) {
```

Comment:
seria bom armazenar esses regex em constantes separadas com um nome descritivo para cada um. Concorda que só `!/\d/.test()` não fica claro o que está sendo testado? Se armazenar numa varivavel, vai fazer vc lembrar o que o regex faz:

```suggestion
  const onlyCharacters = /\d/;
  const onlyNumber = /[a-zA-Z]/;

  if (validPassword.length < 6 || !onlyCharacters.test(validPassword) || !onlyNumber.test(validPassword)) {
```